core.js

```diff
@@ -62,7 +62,11 @@ async function fromTokenizer(tokenizer) {
 	const check = (header, options) => _check(buffer, header, options);
 	const checkString = (header, options) => check(stringToBytes(header), options);
 
-	await tokenizer.peekBuffer(buffer, 0, bytesRead);
+	if (!tokenizer.fileInfo.size) {
+		tokenizer.fileInfo.size = Number.MAX_SAFE_INTEGER;
+	}
```

Comment:
It's not clear to me why we need to do this. Could you add a short code comment?

Reply:
In some of the "level-2" parsers there using conditional based on the file-size:

https://github.com/sindresorhus/file-type/blob/e90035a1b46d3fe9347d90c8414ef9955c5f2701/core.js#L147

https://github.com/sindresorhus/file-type/blob/e90035a1b46d3fe9347d90c8414ef9955c5f2701/core.js#L225

https://github.com/sindresorhus/file-type/blob/e90035a1b46d3fe9347d90c8414ef9955c5f2701/core.js#L866

https://github.com/sindresorhus/file-type/blob/e90035a1b46d3fe9347d90c8414ef9955c5f2701/core.js#L927

In general they reading until the file-type detection completed, as long as the end of file was not reached.

If the file size is not known. By setting the `tokenizer.fileInfo.size = Number.MAX_SAFE_INTEGER;` we keep reading until an end-of-file error is thrown.

Reply:
Could you put something like your last paragraph in a code comment there?